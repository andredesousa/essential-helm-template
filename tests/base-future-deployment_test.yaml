suite: V2 test deployment
values:
  - ../values.yaml
templates:
  - base-future-deployment.yaml
tests:
  - it: should pass all kinds of assertion
    documentIndex: 0
    asserts:
      - hasDocuments:
          count: 2
      - isAPIVersion:
          of: apps/v1
      - isKind:
          of: Deployment
      - isNotNull:
          path: spec.template
      - isNotEmpty:
          path: spec.template.spec.containers[0]
      - equal:
          path: spec.template.spec.containers[0].image
          value: mcr.microsoft.com/azuredocs/aks-helloworld:v1
      - contains:
          path: spec.template.spec.containers[0].ports
          content:
            containerPort: 80
      - matchSnapshot:
          path: spec
