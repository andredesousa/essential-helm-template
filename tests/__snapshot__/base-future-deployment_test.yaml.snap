should have a service document:
  1: |
    ports:
    - port: 80
      protocol: TCP
    selector:
      app: aks-future-api
    type: ClusterIP
